### PR TITLE
EREGCSC-2320 -- Enable active state for selected filter item

### PR DESCRIPTION
**Pull request by PhilR8**
Resolves [EREGCSC-2320](https://jiraent.cms.gov/browse/EREGCSC-2320)

**Description**

The Subject Selector in the left sidebar of the Policy Repository page already has bolded styles added to the subject item when you hover over a subject with the cursor, when you focus the subject item using keyboard navigation, and briefly after you select the subject item.  Once you click away from the subject item, however, the bolded style is removed.

This PR will make that bolded style persist as long as that subject is selected.

**This pull request changes:**

- Adds `--selected` styles to `sidebar-li__button`
- watches `$route.query.subjects` to determine is a selected subject needs the `--selected` style

**Steps to manually verify this change:**

1. Visit Policy Repository at the experimental deployment
2. Add some subjects using the subject selector in the right sidebar
3. Ensure that the bolded style remains even after you click away after selecting the subject
4. Ensure that the bolded style does not interfere with the subject typeahead styles, and is restored once the typeahead input is cleared
5. Ensure that the bolded style is applied to selected subjects when visiting a URL with several subjects already selected

### CODE REVIEW

File: solution/ui/regulations/eregs-vite/src/components/policy-repository/SubjectSelector.vue [hunk 1]
```diff
@@ -127,7 +134,11 @@ const filterResetClick = () => {
             <template v-else>
                 <form>
                     <label for="subjectReduce">Filter the subject list</label>
-                    <input id="subjectReduce" v-model="state.filter" type="text" />
+                    <input
+                        id="subjectReduce"
+                        v-model="state.filter"
+                        type="text"
+                    />
```
Comment (PhilR8) [inline solution/ui/regulations/eregs-vite/src/components/policy-repository/SubjectSelector.vue:141]: prettier formatting change

File: solution/ui/regulations/eregs-vite/src/components/policy-repository/SubjectSelector.vue [hunk 2]
```diff
@@ -144,12 +155,15 @@ const filterResetClick = () => {
...
+                            :class="subjectClasses(subject.id)"
                             :data-name="getSubjectName(subject)"
                             :data-id="subject.id"
                             :data-testid="`add-subject-${subject.id}`"
                             :title="subject.full_name"
                             @click="subjectClick"
-                            v-html="subject.displayName || getSubjectName(subject)"
+                            v-html="
+                                subject.displayName || getSubjectName(subject)
+                            "
```
Comment (PhilR8) [inline solution/ui/regulations/eregs-vite/src/components/policy-repository/SubjectSelector.vue:166]: prettier formatting change

File: solution/ui/regulations/eregs-vite/src/components/policy-repository/__snapshots__/SubjectChip.test.js.snap [hunk 3]
```diff
@@ -6,6 +6,7 @@ exports[`Subject Chip > Renders a Subject Chip 1`] = `
     <div>
       <router-link
         class="subject__chip"
+        data-testid="add-subject-chip-2"
         to="[object Object]"
```
Comment (PhilR8) [inline solution/ui/regulations/eregs-vite/src/components/policy-repository/__snapshots__/SubjectChip.test.js.snap:10]: fixes Vitest failure introduced in a previous story

**Review by thwalker6 — APPROVED** [review]
lgtm

**Pull request merged**